app/services/cache_service.rb

```diff
@@ -33,8 +34,12 @@ def cache_only_updated_listings
         l['Id'] == fresh_listing['Id']
       end
 
-      cache_single_listing(fresh_listing) unless
-        listing_unchanged?(prev_cached_listing, fresh_listing)
+      # TODO: find a differnt way to determin if listing is unchanged. This function seems to always return true.
```

Comment:
Did you test this by modifying a field on a `Listing`, like the name?
Under the hood (i.e. in the Salesforce relational database), the `LastModifiedDate` is a column on the `Listings` table. If we update listing images, that will modify a different table, the `Listing_Images` table, which means `LastModifiedDate` would not change in that scenario.

It is possible to get an understanding of what is happening behind the Salesforce API layer by reading the code available to us in the Salesforce web interface: https://sfhousing--full.sandbox.lightning.force.com/lightning/setup/ApexClasses/page?address=%2F01pU0000001hD1h

Reply:
LastModifiedDate does update in every scenario I have tested. The issue is that the prev_cached_listing and fresh_listing both have the same LastModifiedDate, causing this check to always return true. Thanks for linking that ApexClasses, that is insigtful!

Reply:
I was able to verify that the `listing_unchanged?` method is working correctly, both locally and for the PR app.

## Local

1. make sure `CACHE_SALESFORCE_REQUESTS='true'` is in the local .env file
2. enable caching in dev environment by running `bundle exec rails dev:cache`
3. in `config/environments/development.rb`, change the `config.cache_store` value to `config.cache_store = :memory_store`
4. Update a listing name
5. Open the rails console `bundle exec rails c` and run this snippet
    ```ruby
    Force::ListingService.listings(subset: 'browse').find {|l| l['Id'] == 'a0W4U00000IgshdUAB'}.slice('Id', 'Name', 'LastModifiedDate')
    ```
6. You'll see that the `Name` and `LastModifiedDate` values are the older cached ones

## Heroku App

1. Open the console and start `bash`: https://dashboard.heroku.com/apps/dahlia-webapp-pr-1936/access?web-console=dahlia-webapp-pr-1936
2. Open the rails console `bundle exec rails c` and run this to verify that caching is enabled: `Rails.application.config.action_controller.perform_caching`
3. Run the snippet in the rails console:
    ```ruby
    Force::ListingService.listings(subset: 'browse').find {|l| l['Id'] == 'a0W4U00000IgshdUAB'}.slice('Id', 'Name', 'LastModifiedDate')
    ```
4. You'll see that the `Name` and `LastModifiedDate` values are the older cached ones

Reply:
Awesome! I will take a look.

Reply:
I was able to run this once locally, but wen I ran the snippet, I saw the updated values and not the older ones. Now I cannot run `bundle exec rails c` without getting this error: `Could not find cache store adapter for dalli_store (cannot load such file -- active_support/cache/dalli_store) (RuntimeError)`

Reply:
Did you update config.cache_store in the development config file?

Reply:
I do see this behavior in heroku pr env though. I ran browse and it displays the cached results. I then ran the browse again but this time with the force: true param. The latest changes come through as expected.

Reply:
So we could separate the listing image processing and the force=true calls to the Listing Service.

Or inside `cache_single_listing` check if `listing_unchanged`, before running the Force calls. And then call `process_listing_images`

The most accurate approach would be to check if the Listing_Images is different between prev_cached_listings and fresh_listings.

Something like:

```
def listing_images_unchanged?([prev_cached_listing, fresh_listing])
    prev_cached_listing_images = prev_cached_listing.present? && prev_cached_listing['Listing_Images']
    fresh_listing_images = fresh_listing.present? && fresh_listing['Listing_Images']
    # Compare attributes of each listing image
end
```

Any other ideas on next steps for this piece?

Reply:
Based on my findings, `listing_unchanged?` is working normally. So we can revert all changes made to the `cache_only_updated_listings` in this PR.

Not sure what else we need to do besides reverting, unless I'm missing something?

Reply:
I think we would still need to run the process_images for the Listing Images array. If adding a ListingImage to the array in salesforce does not trigger `listing_unchanged?` to be false, then we will not update the ListingImages for that Listing.

Reply:
> If adding a ListingImage to the array in salesforce does not trigger listing_unchanged? to be false

Oh, did you confirm that? I only mentioned that as a hypothesis, I didn't actually do any manual testing for that behavior.